osquery/tables/system/windows/ntfs_acl_permissions.cpp

```diff
@@ -0,0 +1,207 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <AccCtrl.h>
+#include <Aclapi.h>
+#include <map>
+#include <string>
+#include <vector>
+#include <windows.h>
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/filesystem.hpp>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace alg = boost::algorithm;
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+// helper function to get access mode string
+std::string accessModeToStr(const _ACCESS_MODE& accessMode) {
+  switch (accessMode) {
+  case NOT_USED_ACCESS:
+    return "Not Used";
+  case GRANT_ACCESS:
+    return "Grant";
+  case SET_ACCESS:
+    return "Set";
+  case DENY_ACCESS:
+    return "Deny";
+  case REVOKE_ACCESS:
+    return "Revoke";
+  case SET_AUDIT_SUCCESS:
+    return "Set Audit Success";
+  case SET_AUDIT_FAILURE:
+    return "Set Audit Failure";
+  }
+  return "";
+}
+
+// helper function to build access string from permission bit mask
+std::string accessPermsToStr(const unsigned long pmask) {
+  std::vector<std::string> permList;
+  const std::map<unsigned long, std::string> permVals = {
```

Comment:
Same here, maybe make this a statically defined `const std::map` variable in the global namespace, so that it's faster to access and doesn't need to be redefined with every invocation.